services/frontend/src/utils/models/Project.js

```diff
@@ -26,7 +30,7 @@ export default class Project {
     /* Normalizes the list of projects that the API returns into a map of {ID -> project},
      * with the skills processed to just their IDs, for appropriate use in the Redux store. */
     static normalizeApiResultsForRedux(projects = []) {
-        return projects.reduce((acc, project) => {
+        const processProject = (project) => {
```

Comment:
Feels like this function could be outside of `normalizeApiResultsForRedux` due to possibly being reusable, instead of being created and then immediately called with arrayToObject.

Not sure if possible but it could maybe be setup as something like `processWithSkills` that way you could make a few of these (`processWithX`) and not re-write them in every normalize function.

Reply:
My goto rule of thumb is that unless something is _actually_ needed somewhere else, just keep it as local as possible; this is why I do things like not refactoring components out into their own files until they're actually needed somewhere else. YAGNI and all that.

Now, to address your first point about it being created and then immediately called, do you think you would have made that same comment if I had just inlined the function _inside_ `arrayToObject`? Food for thought.

As for your second point about having a suite of `processWith` functions, I'll invoke YAGNI again here. If the processing functions are only ever used (currently) in their respective normalizing functions, then there is no reason to refactor them out and make them 'reusable', until such time comes that the opportunity presents itself. As such, no, I don't think refactoring them out _right now_ is of any use to us.

Reply:
Fair points, it's likely I wouldn't have mentioned it if it was in inside the arrayToObject but I feel that leans more into how obvious the functionality can be exported. It's harder to realize reusable code when it's deeply nested.

Also yep, separating out those functions right now doesn't achieve a whole lot.